### DISCUSSION TITLE
Object of class GraphQL\Type\Definition\FieldDefinition could not be converted to number

### DISCUSSION THREAD

**Discussion by mitulmlakhani** [rebing/graphql-laravel#838] [Q&A]
### Versions:
- graphql-laravel Version: ^7.2
- Laravel Version: 7.0 
- PHP Version: 7.3.12

### Description:
Interface fields sharing on input type throws an error.

Please describe in detail the nature of the bug, code samples, etc.
Error:
![image](https://user-images.githubusercontent.com/24645690/138833776-266bf04e-3079-4c9c-9f59-7433a270ef3d.png)

Interface: 
![image](https://user-images.githubusercontent.com/24645690/138832516-83692e2b-c915-4cee-9bec-0b42dcc04bca.png)

Type: 

![image](https://user-images.githubusercontent.com/24645690/138832589-fc02763c-e81c-454d-a9cb-c92799371d05.png)

**Comment by mfn** [comment]
Sorry I really can't say from the code I see.

> … could not be converted to number

I usually see this when a the underlying webonxy GraphQL library expects a "string array key" somewhere but receives just an integer, often due to an invalid field spec. I.e. instead something like this (just made up)
```php
'roleId' => Type::id(),
```
That isn't the case according to your screenshots, but just giving some pointers maybe this or a similar error happens somewhere else not visible here.

---
If you provide a standalone project on github to check out with the code not working and some basic instruction how to run it, I can offer to take a look there. It's usually easier with so many definitions required in GraphQL.
